inc/saiars.h

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiars.h
+ *
+ * @brief   This module defines SAI interface for adaptive routing and switching
+ */
+
+#if !defined (__SAIARS_H_)
+#define __SAIARS_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIARS SAI - Adaptive Routing and Switching specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Adaptive routing and switching path (re)assignment mode
+ */
+typedef enum _sai_ars_mode_t
+{
+    /** Per flow-let quality based path (re)assignment */
+    SAI_ARS_MODE_FLOWLET_QUALITY,
+
+    /** Per flow-let random path (re)assignment */
+    SAI_ARS_MODE_FLOWLET_RANDOM,
+
+    /** Per packet quality based path (re)assignment */
+    SAI_ARS_MODE_PER_PACKET_QUALITY,
+
+    /** Per packet random path (re)assignment */
+    SAI_ARS_MODE_PER_PACKET_RANDOM,
+
+    /** Fixed path assignment */
+    SAI_ARS_MODE_FIXED,
+
+} sai_ars_mode_t;
+
+/**
+ * @brief Attribute id for adaptive routing and switching
+ */
+typedef enum _sai_ars_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_ARS_ATTR_START,
+
+    /**
+     * @brief ARS path assignment mode
+     *
+     * @type sai_ars_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_ARS_MODE_FLOWLET_QUALITY
+     */
+    SAI_ARS_ATTR_FLOWLET_MODE = SAI_ARS_ATTR_START,
```

Comment:
Per type above the mode can be one of 3 options. Why should this attribute be named by one of them ? Isn't SAI_ARS_ATTR_ MODE enough ?